mlflow/system_metrics/system_metrics_monitor.py

```diff
@@ -0,0 +1,128 @@
+"""Class for monitoring system stats."""
+
+import logging
+import os
+import threading
+
+from mlflow.system_metrics.metrics.cpu_monitor import CPUMonitor
+from mlflow.system_metrics.metrics.gpu_monitor import GPUMonitor
+
+_logger = logging.getLogger(__name__)
+
+
+class SystemMetricsMonitor:
+    """Class for monitoring system stats.
+
+    This class is used for pulling system metrics and logging them to MLflow. Calling `start()` will
+    spawn a thread that logs system metrics periodically. Calling `finish()` will stop the thread.
+    Logging is done on a different frequency from pulling metrics, so that the metrics are
+    aggregated over the period. Users can change the logging frequency by setting
+    `$MLFLOW_SYSTEM_METRICS_SAMPLING_INTERVAL` and `$MLFLOW_SYSTEM_METRICS_SAMPLES_BEFORE_LOGGING`
+    environment variables.
+
+    Args:
+        mlflow_run: an 'mlflow.entities.run.Run' instance, which is used to bootstrap system metrics
+            logging with the MLflow tracking server.
+        sampling_interval: float, default to 0.5. The interval (in seconds) at which to pull system
+            metrics.
+        samples_to_aggregate: int, default to 30. The number of samples to aggregate before logging.
+    """
+
+    def __init__(self, mlflow_run, sampling_interval=10, samples_before_logging=1):
+        from mlflow.utils.autologging_utils import BatchMetricsLogger
+
+        # Instantiate default monitors.
```

Comment:
Are the disk and network monitors configurable here? 
Can users specify which ones to initialize?

Reply:
This is actually a complex part - by default it's better that we just collect a set of system metrics we think are useful, like wandb or neptune.ai's approach. In this PR disk and network are not included because every system metric shows up in "metrics view", a long list makes it super messy, so we only include CPU/GPU here. 

Back to the question of configurable monitoring, maybe the simplest way is for each type of metrics, we create an argument, e.g, "enable_disk_monitoring=True" in the constructor. Still it's not fully configurable because our code decides what kinds of metrics are collected within each monitor, e.g, "disk_percentage" is included but `disk_free` is not.